File: Rust/prototype/src/structs/foo.rs
use chrono::{DateTime, Utc};

use crate::structs::example_data::ExampleData;

pub struct Foo {
    pub name: String,
    pub identifier: String,
    pub timestamp: DateTime<Utc>,
}

impl ExampleData for Foo {
    fn get_message(&self) -> String {
        return format!("{} [{}] - Foo says \"Hello, {}!\"", self.identifier, self.timestamp, self.name);
    }
}

impl Foo {
    pub fn new(name: &str, identifier: &str, timestamp: DateTime<Utc>) -> Self {
        return Self {
            name: String::from(name),
            identifier: String::from(identifier),
            timestamp,
        };
    }
}

File: Java/ModularService/ModularLibrary/src/main/java/com/hydron/sample/modularlibrary/pojo/Foo.java
package com.hydron.sample.modularlibrary.pojo;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.Instant;
import java.util.Arrays;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@ToString
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Foo implements ExampleData {
    @JsonProperty
    private String[] names;

    @JsonProperty
    private String identifier;

    @JsonProperty
    private Instant timestamp;

    @JsonProperty
    private static final String text = "%s [%s] - Foo says \"Hello %s!\"";

    @Override
    public String getMessage() {
        return String.format(text, this.identifier, this.timestamp, Arrays.toString(this.names));
    }
}
